registry/client/repository.go

```diff
@@ -21,6 +21,85 @@ import (
 	"github.com/docker/distribution/registry/storage/cache/memory"
 )
 
+// Namespace provides an interface for calling Repositories.  It's intended to be a slimmed down
+// version of distribution.Namespace since that interface implements additional functions which are
+// currently not supported through namespaces.
+type Namespace interface {
```

Comment:
Just call this `Catalog`. It will be confusing to mix this up with `distribution.Namespace`. We'll fix up the naming later.